src/Symfony/Component/DependencyInjection/Dumper/YamlDumper.php

```diff
@@ -247,10 +249,11 @@ private function dumpCallable($callable)
      */
     private function dumpValue($value)
     {
-        if ($value instanceof IteratorArgument) {
-            $value = array('=iterator' => $value->getValues());
-        } elseif ($value instanceof ClosureProxyArgument) {
-            $value = array('=closure_proxy' => $value->getValues());
+        if ($value instanceof IteratorArgument || $value instanceof ClosureProxyArgument) {
+            $argument = clone $value;
+            $argument->setValues($this->dumpValue($argument->getValues()));
```

Comment:
I'm sure there are ways to fix this with the current design.
I still think the "TaggedValue" proposal could be a good idea to try :)